### PR TITLE
[E2E] Upgrade ginkgo to v2

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Ginkgo 2.0 is a major release that adds substantial new functionality and removes/moves existing functionality.

There are many new functionality, just as [spec-decorators](https://onsi.github.io/ginkgo/MIGRATING_TO_V2#spec-decorators), [Spec Labels](https://onsi.github.io/ginkgo/MIGRATING_TO_V2#spec-labels), [DeferCleanup](https://onsi.github.io/ginkgo/MIGRATING_TO_V2#defercleanup) and so on, we decided to try and embrace it.

The upgrade guide refer to [here](https://onsi.github.io/ginkgo/MIGRATING_TO_V2#upgrading-to-ginkgo-20).

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/suite_test.go [hunk 1]
```diff
@@ -109,7 +104,7 @@ var _ = ginkgo.BeforeSuite(func() {
 
 	err = setupTestNamespace(testNamespace, kubeClient)
 	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
-}, TestSuiteSetupTimeOut.Seconds())
+})
```
Comment (RainbowMango) [inline test/e2e/suite_test.go:107]: What's the timeout now?

**Review by RainbowMango — COMMENTED** [review]
LGTM.
Just one question.

File: test/e2e/suite_test.go (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply test/e2e/suite_test.go:107]: Maybe it's the reason: https://onsi.github.io/ginkgo/MIGRATING_TO_V2#timeout-behavior

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Pull request merged**